### PR TITLE
Add single retry on Connection errors

### PR DESCRIPTION
**Pull request by asuresh4**
Add a single retry on Connection errors.

Similar update as https://github.com/signalfx/signalfx-python/pull/104

### CODE REVIEW

**Comment by asuresh4** [comment]
```
(jaeger-client-python) ❯ make test
py.test -s --tb short --cov-config .coveragerc --cov jaeger_client --cov crossdock tests --cov-report term-missing --cov-report xml --junitxml junit.xml --benchmark-skip
============================================================================================================================= test session starts ==============================================================================================================================
platform darwin -- Python 3.7.9, pytest-3.7.4, py-1.10.0, pluggy-1.0.0
benchmark: 3.1.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /Users/akashsuresh/work/jaeger-client-python, inifile: setup.cfg
plugins: benchmark-3.1.1, cov-2.5.1, tornado-0.8.1, localserver-0.5.0, timeout-1.3.1
timeout: 60.0s
timeout method: signal
timeout func_only: False
collected 245 items

tests/test_TUDPTransport.py ......
tests/test_api.py .........................
tests/test_codecs.py .........................
tests/test_config.py ..........................
tests/test_crossdock.py sssssssss..
tests/test_ioloop_util.py ......
tests/test_local_agent_net.py ..
tests/test_metrics.py .........
tests/test_noop_tracer.py ..
tests/test_prometheus.py ....
tests/test_rate_limiter.py .
tests/test_reporter.py .............
tests/test_sampler.py ................................
tests/test_senders.py ................
tests/test_span.py .........
tests/test_span_context.py ...
tests/test_thrift.py .........
tests/test_throttler.py .....
tests/test_tracer.py .......................
tests/test_tracer_benchmark.py sssssss
tests/test_utils.py ...........

-------------------------------------------------------------------------------------------------- generated xml file: /Users/akashsuresh/work/jaeger-client-python/junit.xml --------------------------------------------------------------------------------------------------

---------- coverage: platform darwin, python 3.7.9-final-0 -----------
Name                                      Stmts   Miss Branch BrPart  Cover   Missing
-------------------------------------------------------------------------------------
crossdock/__init__.py                         4      0      0      0   100%
crossdock/server/__init__.py                  0      0      0      0   100%
crossdock/server/constants.py                 1      1      0      0     0%   15
crossdock/server/endtoend.py                 56     23      6      1    55%   70-101, 108, 112, 116-117, 126-134, 144, 139->144
crossdock/server/serializer.py               69     69     36      0     0%   15-122
crossdock/server/server.py                  129    129     16      0     0%   16-229
crossdock/server/thriftrw_serializer.py      12     12      4      0     0%   16-40
crossdock/setup_crossdock.py                  5      5      0      0     0%   18-25
jaeger_client/TUDPTransport.py               21      0      2      0   100%
jaeger_client/__init__.py                    13      0      0      0   100%
jaeger_client/codecs.py                     196      4    106      5    97%   70, 78, 114, 118, 69->70, 77->78, 113->114, 117->118, 132->130
jaeger_client/config.py                     204      6     46      2    97%   161, 260-265, 434, 413->422, 433->434
jaeger_client/constants.py                   26      0      0      0   100%
jaeger_client/ioloop_util.py                 44      0     20      1    98%   51->exit
jaeger_client/local_agent_net.py             33      0      4      0   100%
jaeger_client/metrics/__init__.py             4      0      0      0   100%
jaeger_client/metrics/metrics.py             57      0     16      0   100%
jaeger_client/metrics/prometheus.py          35      0     10      0   100%
jaeger_client/rate_limiter.py                28      0      4      0   100%
jaeger_client/reporter.py                   151      4     36      1    97%   99, 197-199, 98->99
jaeger_client/sampler.py                    301      9     70      3    97%   420, 488-495, 380->388, 382->388, 505->508
jaeger_client/senders.py                    134      3     22      1    97%   98, 238-239, 135->138
jaeger_client/span.py                       108      1     34      3    97%   110, 107->110, 153->152, 160->159
jaeger_client/span_context.py                29      0      2      0   100%
jaeger_client/thrift.py                      88      3     36      2    96%   62, 65-66, 26->30, 61->62
jaeger_client/throttler.py                  122      7     28      4    91%   91-93, 143-144, 184-185, 68->72, 173->175, 183->184, 186->189
jaeger_client/tracer.py                     154      3     54      1    98%   272, 297, 310, 296->297
jaeger_client/utils.py                       60      5     16      1    89%   73-74, 89-90, 107, 61->63
-------------------------------------------------------------------------------------
TOTAL                                      2084    284    568     25    86%
```

**Comment by asuresh4** [comment]
cc @owais

File: jaeger_client/senders.py [hunk 1]
```diff
@@ -221,6 +226,14 @@ def send(self, batch):
...
+            self._post(data, headers)
         except Exception as e:
             raise_with_value(e, 'POST to jaeger_endpoint failed: {}'.format(e))
+
+
+    def _post(self, data, headers):
+        try:
+            self.client.post(url=self.url, headers=headers, data=data)
+        except ConnectionError:
+            self._reconnect()
```
Comment (owais) [inline jaeger_client/senders.py:239]: we should log a msg that says the export is being retried.

**Review by owais — APPROVED** [review]

File: jaeger_client/senders.py (hunk 1 above)
Comment (asuresh4) [reply jaeger_client/senders.py:239]: Updated.

**Pull request merged**